doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,509 @@
+=============================
+Basic notions and conventions
+=============================
+
+This section provides some basic vocabulary and conventions to read
+this manual.
+
+We start by explaining the syntax and lexical conventions used in this
+manual.
+
+Then, we explain the basic notions of the |Coq| language.  In
+particular, we provide a short glossary defining the most commonly
+used vocabulary.  Other glossary terms are defined throughout the
+manual and the reader may refer to the :ref:`glossary index
+<glossary_index>` for a more complete reference.
+
+Syntax and lexical conventions
+------------------------------
+
+Syntax conventions
+~~~~~~~~~~~~~~~~~~
+
+|Coq| contains an extensible parsing engine.  The extensions to the
+|Coq| syntax can come from either plugins, or the :ref:`notation
+system <syntax-extensions-and-notation-scopes>`.  Furthermore, some
+parsing rules are only activated in certain contexts
+(:ref:`interactive proof mode <proofhandling>`, :ref:`custom entries
+<custom-entries>`...).  The syntax described in this documentation is
+equivalent to the rules implemented in the |Coq| parser, but has been
+heavily edited to improve readability and presentation.
```

Comment:
Think I'd reorder the sentences something like this (with a little rewording):

The syntax described in this documentation is
equivalent that accepted by the |Coq| parser, but the grammar has been
heavily edited to improve readability and presentation.

Coq uses an extensible parser. Plugins and the ref:`notation
system <syntax-extensions-and-notation-scopes>` can extend the syntax at run time.
Some notation are predefined by default.  The documented grammar doesn't
include such syntax extensions.

> Furthermore, some
parsing rules are only activated in certain contexts
(:ref:`interactive proof mode <proofhandling>`, :ref:`custom entries
<custom-entries>`...).

Yeah, some things are semantically restricted.  Do you think we need say that
explicitly here?  We've not introduced interactive proof mode or custom entries
and they seem too detailed for here.

WDYT?

Reply:
Looks good.